### ISSUE TITLE
API / Unable to POST setstate

### ISSUE THREAD

**Issue by Bbillyben** [Forceu/barcodebuddy#199]
Hi There,

I'm developping a java app to concatenate barcodes read by a hand scanner and send it to barcodebuddy via api.

I'm able to read info (`api/system/info`), get current state (`api/state/getmode`), even send barcode (`api/action/scan`).

But I'm not able to send a state change (`api/state/setmode`) :

my code to interact with the api is like :

```
    public static int setState(String bcbUrl, String bcbAPI, int state){
        System.out.println("query info : "+bcbUrl + " / " + bcbAPI + " / state : " + state);
        try {
            Response response = new JdkRequest(bcbUrl)
                    .uri().path("/api/state/setmode")
                    //.queryParam("state", state)
                    .back()
                    .method(Request.POST)
                    .body()
                    .formParam("state", "STATE_PURCHASE")
                    .back()
                    .header("accept", "application/json")
                    .header("Content-Type", "multipart/form-data")
                    .header("BBUDDY-API-KEY", bcbAPI)
                    .fetch();
            int status = response.status();
            String body = response.body();
            System.out.println("set state resp : " + body);
            return status;
        } catch (IOException ex) {
            return 0;
        }
    }

```

it use `com.jcabi.http` library

but the only answer i receive is a 400 code:
```
{
   "data":null,
   "result":{
      "result":"Invalid state provided",
      "http_code":400
   }
}
```

 but it doesn't seems tied to the platform or code, as i've tried with a REST client, and same result,

tried a curl command from a debian machine :

`curl -X GET "https://mybcb.server.com/api/state/setmode" -H "accept: application/json" -H "BBUDDY-API-KEY: [[apikey]]" -H "Content-Type: multipart/form-data" -F "state=2"`

and again same result.

I event try to replace the int with "STATE_PURCHASE"

but nothing seems working at all.

It feels like I haven't activated something in bcb

Is there anything I missed?

**Comment by Bbillyben** [comment]
Ok, as usual, As i got decided to post, I found .....

I read the log in the docker container (`/config/log/nginx/error.log`)

see :

`PHP message: PHP Warning: Missing boundary in multipart/form-data POST data [...]`


after a quick search, the solution is to remove :

`.header("Content-Type", "multipart/form-data")`


... less than an hour after a day on it... (rtfL)

**Comment by Forceu** [comment]
Yes, the API uses `application/x-www-form-urlencoded` instead of multipart. Glad to hear it is working now!

**Comment by Bbillyben** [comment]
thanks Forceu, 

That's the clue! it works great with this header!

 
The API helper confused me as it show the "Content-Type: multipart/form-data" header, and the 'http request' uneducated person I'am was stopped !
